docs/rfcs/0003-etag-changes.md

```diff
@@ -24,54 +24,153 @@ While the current Etag implementation allows for easily adding and removing Etag
+    Api1 ->> User: Return updated resource
+```
+
+While this would work, there's a lot of additional complexity over the distributed cache, while providing the same benefits along with potential race conditions.  As such, while this has been mapped out for posterity, this option is being discarded.
 
 ### Generating the checksum
 
-If the checksum is now being generarted upfront instead of using the GET response to generate an Etag, there are some other ways of doing this:
+If the checksum is now being generated upfront instead of using the GET response to generate an Etag, there are some other ways of doing this:
 
 #### Using the hash
 This is the same way that it currently works, which is fine, but could mean there are differences between a checksum generated from a hierarchical PUT/POST and a flat PUT/POST
 
 There's an issue with using the file to generate the hash in that PUT/POST can occur on the hierarchical path and the flat path which has different styles of request body.  This means that potentially updating a resource could have 2 different file hashes if using the request.  This can also happen with variations of `publicId`, `slug` and `parent`. Given these are ultimately the same request, it would be better for a single value to control the Etag on all of these requests and responses. This could be done using the response instead of the request, but there are issues with this around properties such as `ingesting` which will change based on assets. However, this would likely make for a better value when PATCH is integrated.
 
-##### Questions
-
-- Is it worth worrying about differences in the request or response?  It feels like just using the latest PUT/POST would be unique enough
+Additionally, the PUT/POST response is modified at a later date byt the background handler if there are assets attached to the manifest.  This would mean that Etags would not be available for manifests with assets until after processing was completed.  The full process would look like this:
+
+```mermaid
+sequenceDiagram
+    actor User 
+    User ->> Api: Update presentation resource
+    Api ->> Api: Process resource
+    alt No assets on manifest
+        Api ->> Database: Update Etag
+    else Assets on manifest
+        participant Background as Background Handler
+        Background ->> Background: Update manifest
+        Background ->> Database: Update Etag
+    end
+    Api ->> User: Return updated resource
+```
  
 #### Leveraging S3
 
-- S3 has the ability to generate a file hash we could take advantage of
+S3 automatically generates a file hash we could take advantage of for using as an Etag value
+
+  - This file hash is pre-generated and would be easy to retreive
 - This has issues with storage collections, as these aren't stored in S3 - we could use a different system for generating this hash, but then that means there is a different code journey in this specific scenario, which has caused issues with other parts of the presentation API
```

Comment:
> a different code journey in this specific scenario

I don't think this will have any bearing on the outcome but this doesn't seem a different code journey, instead just a different implementation:

```cs
public class Something(IEtagGenerator etagGenerator)
{
  public void Processor(Collection collection) 
  {
    var etag = etagGenerator.GenerateEtag(collection);
  }
}

public class StorageCollectionGen : IEtagGenerator { }

public class NotStorageCollectionGen : IEtagGenerator { }
```